DataFormats/Track/interface/alpaka/PixelTrackUtilities.h

```diff
@@ -0,0 +1,199 @@
+#ifndef DataFormats_Track_PixelTrackUtilities_h
+#define DataFormats_Track_PixelTrackUtilities_h
+
+#include "DataFormats/TrackerCommon/interface/SimplePixelTopology.h"
+#include "DataFormats/Track/interface/PixelTrackDefinitions.h"
+#include "DataFormats/Track/interface/PixelTrackLayout.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  // Methods that operate on View and ConstView of the TrackSoA, and cannot be class methods.
+  template <typename TrackerTraits>
+  struct TracksUtilities {
+    using TrackSoAView = typename TrackSoA<TrackerTraits>::template TrackSoAHeterogeneousLayout<>::View;
+    using TrackSoAConstView = typename TrackSoA<TrackerTraits>::template TrackSoAHeterogeneousLayout<>::ConstView;
+    using hindex_type = typename TrackSoA<TrackerTraits>::hindex_type;
+
+    // State at the Beam spot
+    // phi,tip,1/pt,cotan(theta),zip
+    ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE static float charge(const TrackSoAConstView &tracks, int32_t i) {
```

Comment:
Is this function actually used ?

In principle, `std::copysign()` should not work in _e.g._ CUDA device code, because it's not `constexpr`.